llvm-external-projects/iree-dialects/test/Dialect/iree_vector_ext/canonicalize.mlir

```diff
@@ -0,0 +1,43 @@
+// RUN: iree-dialects-opt --canonicalize --split-input-file %s | FileCheck %s
+
+// CHECK-LABEL: @to_simt_to_simd_fold
+func.func @to_simt_to_simd_fold(%simd: vector<64x64xf32>) -> vector<64x64xf32> {
+  // Both to_simt and to_simd should be dce-ed after folding.
+  // CHECK-NOT: iree_vector_ext.to_simt
+  %simt = iree_vector_ext.to_simt %simd : vector<64x64xf32> -> vector<4x4x4xf32>
```

Comment:
NIT: For the very first one, would be nice to show that the function becomes something like:
@to_simt_to_simd_fold(%arg0)
func.return %arg0

Reply:
Added